test/val/val_id_test.cpp

```diff
@@ -2103,7 +2103,271 @@ OpFunctionEnd
+  string spirv = kGLSL450MemoryModel + opAccessChainSpirvSetup + R"(
+%float_entry = OpInBoundsAccessChain %_ptr_Private_float %void %int_0 %int_1
+OpReturn
+OpFunctionEnd
+  )";
+  CompileSuccessfully(spirv);
+  EXPECT_EQ(SPV_ERROR_INVALID_ID, ValidateInstructions());
+  EXPECT_THAT(getDiagnosticString(),
+              HasSubstr("The Base <id> '1' in OpInBoundsAccessChain instruction must "
+                        "be a pointer."));
+}
+
+// Invalid. The base type of OpInBoundsAccessChain must be a pointer.
+TEST_F(ValidateIdWithMessage, OpInBoundsAccessChainBaseTypeNonPtrVariableBad) {
+  string spirv = kGLSL450MemoryModel + opAccessChainSpirvSetup + R"(
+%entry = OpInBoundsAccessChain %_ptr_Private_float %_ptr_Private_float %int_0 %int_1
+OpReturn
+OpFunctionEnd
+  )";
+  CompileSuccessfully(spirv);
+  EXPECT_EQ(SPV_ERROR_INVALID_ID, ValidateInstructions());
+  EXPECT_THAT(getDiagnosticString(),
+              HasSubstr("The Base <id> '8' in OpInBoundsAccessChain instruction must "
+                        "be a pointer."));
+}
+
+// Invalid: The storage class of Base and Result do not match.
+TEST_F(ValidateIdWithMessage,
+       OpInBoundsAccessChainResultAndBaseStorageClassDoesntMatchBad) {
+  string spirv = kGLSL450MemoryModel + opAccessChainSpirvSetup + R"(
+%entry = OpInBoundsAccessChain %_ptr_Function_float %my_matrix %int_0 %int_1
+OpReturn
+OpFunctionEnd
+  )";
+  CompileSuccessfully(spirv);
+  EXPECT_EQ(SPV_ERROR_INVALID_ID, ValidateInstructions());
+  EXPECT_THAT(getDiagnosticString(),
+              HasSubstr("The result pointer storage class and base pointer "
+                        "storage class in OpInBoundsAccessChain do not match."));
+}
+
+// Invalid. The base type of OpInBoundsAccessChain must point to a composite object.
+TEST_F(ValidateIdWithMessage, OpInBoundsAccessChainBasePtrNotPointingToCompositeBad) {
+  string spirv = kGLSL450MemoryModel + opAccessChainSpirvSetup + R"(
+%entry = OpInBoundsAccessChain %_ptr_Private_float %my_float_var %int_0
+OpReturn
+OpFunctionEnd
+  )";
+  CompileSuccessfully(spirv);
+  EXPECT_EQ(SPV_ERROR_INVALID_ID, ValidateInstructions());
+  EXPECT_THAT(getDiagnosticString(),
+              HasSubstr("OpInBoundsAccessChain reached non-composite type while "
+                        "indexes still remain to be traversed."));
+}
+
+// Invalid. No Indexes passed to OpInBoundsAccessChain
+TEST_F(ValidateIdWithMessage, OpInBoundsAccessChainMissingIndexesBad) {
+  string spirv = kGLSL450MemoryModel + opAccessChainSpirvSetup + R"(
+%entry = OpInBoundsAccessChain %_ptr_Private_float %my_float_var
+OpReturn
+OpFunctionEnd
+  )";
+  CompileSuccessfully(spirv);
+  EXPECT_EQ(SPV_ERROR_INVALID_ID, ValidateInstructions());
+  EXPECT_THAT(getDiagnosticString(),
+              HasSubstr("No Indexes were passes to OpInBoundsAccessChain."));
+}
+
+// Invalid: 256 indexes passed to OpInBoundsAccessChain. Limit is 255.
+TEST_F(ValidateIdWithMessage, OpInBoundsAccessChainTooManyIndecesBad) {
```

Comment:
Let's use "Indexes" or "Indices" consistently.  "Indeces" is not a word. :-)

Likely affects the existing tests too.